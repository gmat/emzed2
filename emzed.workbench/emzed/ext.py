import pkg_resources

for ep in pkg_resources.iter_entry_points("emzed_plugin", name="package"):
    print ep
    package = ep.load()
    exec "%s=package" % package.__name__
    del package
    del ep
del pkg_resources

