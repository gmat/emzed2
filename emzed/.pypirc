[distutils]
index-servers = 
  local

[local]
repository: http://127.0.0.1:8080
username:uschmitt
password:pillepalle

