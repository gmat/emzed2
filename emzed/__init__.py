# encoding:latin-1

from version import version as __version__


