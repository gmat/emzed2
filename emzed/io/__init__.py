
from load_utils import loadCSV, loadTable, loadPeakMap
from store_utils import storeCSV, storeTable, storePeakMap

try:
    del store_utils
except:
    pass
try:
    del load_utils
except:
    pass
