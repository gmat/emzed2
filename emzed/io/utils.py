

def _prepare_path(path, extensions):

    import sys
    if isinstance(path, unicode):
        path = path.encode(sys.getfilesystemencoding())
    elif path is None:
        from .. import gui
        path = gui.askForSingleFile(extensions=extensions)
    return path
