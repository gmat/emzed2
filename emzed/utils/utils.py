import emzed.core.data_types

def toTable(colName, iterable,  format_="", type_=None, title="", meta=None):
    return emzed.core.data_types.Table.toTable(colName, iterable, format_, type_, title, meta)

toTable.__doc__ = emzed.core.data_types.Table.toTable.__doc__

def formula(mf):
    """
    Creates formula object which allows addition and subtraction:

    .. pycon::

       import ms   !onlyoutput
       mf1 = ms.formula("H2O")
       mf2 = ms.formula("NaOH")
       mf3 = mf1 + mf2
       print str(mf3)
       print str(mf3 - mf1)

    Mass calculation is supported too:

    .. pycon::

       print mf1.mass()

    If you need some internal representation, you can get a dictionary.
    Keys are pairs of *(symbol, massnumber)*, where *massnumber = None*
    refers to the lowest massnumber. Values of the dictionary are counts:

    .. pycon::

       print mf1.asDict()
       mixed = ms.formula("[13]C2[14]C3")
       print mixed.asDict()


    """
    from emzed.core.chemistry import MolecularFormula
    return MolecularFormula(mf)

def addmf(formula0, *formulas):
    """
    Combines molecular formulas by addition and subtraction:

    .. pycon::

       import ms !onlyoutput
       print ms.addmf("H2O", "COOH")
       print ms.addmf("H2O", "COOH", "NaCl")

    A leading minus sign subtracts the formula following this sign:

    .. pycon::

       print ms.addmf("H2O2", "-H2O")
       print ms.addmf("H2O", "COOH", "NaCl", "-H2O2")
       print ms.addmf("(CH2)7COOH", "-C7")

    """

    mf0 = formula(formula0)
    for f in formulas:
        if f.startswith("-"):
            mf0 -= formula(f[1:])
        else:
            mf0 += formula(f)
    return str(mf0)


###


def openInBrowser(urlPath):
    """
    opens *urlPath* in browser, eg:

    .. pycon::
        ms.openInBrowser("http://emzed.biol.ethz.ch") !noexec

    """
    from PyQt4.QtGui import QDesktopServices
    from PyQt4.QtCore import QUrl
    import os.path

    url = QUrl(urlPath)
    scheme = url.scheme()
    if scheme not in ["http", "ftp", "mailto"]:
        # C:/ or something simiar:
        if os.path.splitdrive(urlPath)[0] != "":
            url = QUrl("file:///"+urlPath)
    ok = QDesktopServices.openUrl(url)
    if not ok:
        raise Exception("could not open '%s'" % url.toString())


def _recalculateMzPeakFor(postfix):
    def calculator(table, row, name, postfix=postfix):

        mzmin = table.get(row, "mzmin"+postfix)
        mzmax = table.get(row, "mzmax"+postfix)
        rtmin = table.get(row, "rtmin"+postfix)
        rtmax = table.get(row, "rtmax"+postfix)
        pm    = table.get(row, "peakmap"+postfix)
        mz = pm.representingMzPeak(mzmin, mzmax, rtmin, rtmax)
        return mz if mz is not None else (mzmin+mzmax)/2.0
    return calculator

def _hasRangeColumns(table, postfix):
    return all([table.hasColumn(n + postfix) for n in ["rtmin", "rtmax",
                                                 "mzmin", "mzmax", "peakmap"]])

def recalculateMzPeaks(table):
    #TODO: tests !
    """Adds mz value for peaks not detected with centwaves algorithm based on
       rt and mz window: needed are columns mzmin, mzmax, rtmin, rtmax and
       peakmap mz, postfixes are automaticaly taken into account"""
    postfixes = [ "" ] + [ "__%d" % i for i in range(len(table._colNames))]
    for postfix in postfixes:
        if _hasRangeColumns(table, postfix):
            mz_col = "mz" + postfix
            if table.hasColumn(mz_col):
                table.replaceColumn(mz_col, _recalculateMzPeakFor(postfix),
                                    format="%.5f", type_=float)
            else:
                table.addColumn(mz_col, _recalculateMzPeakFor(postfix),
                                format="%.5f", type_=float)

def startfile(path):
    import sys, os
    if sys.platform == "win32":
        path = path.replace("/", "\\") # needed for network paths
    return os.startfile(path)

