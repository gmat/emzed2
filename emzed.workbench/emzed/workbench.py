def run():
    print 42

def main():
    import os
    print os.system("pip install -i http://127.0.0.1:3142/root/dev/+simple emzed.core")
    import emzed.core
    print emzed.core



