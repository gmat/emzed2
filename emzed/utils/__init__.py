from utils import *
from isotope_calculator import isotopeDistributionTable, plotIsotopeDistribution
from formula_generator  import formulaTable

from load_utils import loadCSV, loadTable, loadPeakMap

from feature_detectors import runCentwave, runMatchedFilters, runMetaboFeatureFinder

from metlin import matchMetlin

from integration import integrate
