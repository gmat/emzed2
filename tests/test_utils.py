import emzed.utils
import emzed.io
import os.path as osp


def test_formula():
    mf1 = emzed.utils.formula("H2O")
    mf2 = emzed.utils.formula("CH2O")
    assert str(mf1+mf2) == "CH4O2"
    mf3 = mf1 + mf2 - mf2
    assert str(mf3) == "H2O"


def test_recalc_peaks(path):
    from_ = path(u"data/SHORT_MS2_FILE.mzData")
    ds = emzed.io.loadPeakMap(from_)
    t = emzed.utils.toTable("mzmin", [0.0])
    t.addColumn("mzmax", [1000.0])
    t.addColumn("rtmin", [0.0])
    t.addColumn("rtmax", [1000.0])
    t.addColumn("peakmap", [ds])

    emzed.utils.recalculateMzPeaks(t)
    t.info()
    to_be = 807.354
    diff = abs(t.mz.values[0] - to_be)
    assert diff < 1e-3, to_be

def test_load_map(path, tmpdir):
    from_ = path(u"data/SHORT_MS2_FILE.mzData")
    ds = emzed.io.loadPeakMap(from_)
    assert osp.basename(ds.meta.get("source")) ==  osp.basename(from_)

    # with unicode
    emzed.io.storePeakMap(ds, tmpdir.join("utilstest.mzML").strpath)
    ds2 = emzed.io.loadPeakMap(tmpdir.join("utilstest.mzML").strpath)
    assert len(ds)==len(ds2)

    # without unicode
    emzed.io.storePeakMap(ds2, tmpdir.join("utilstest.mzData").strpath)
    ds3 = emzed.io.loadPeakMap(tmpdir.join("utilstest.mzData").strpath)

    assert len(ds)==len(ds3)
    assert ds3.uniqueId() == ds2.uniqueId()


def test_merge_tables():
    t1 = emzed.utils.toTable("a", [1,2])
    t2 = emzed.utils.mergeTables([t1, t1])
    assert len(t2) == 2*len(t1)


def test_stack_tables():
    t1 = emzed.utils.toTable("a", [1,2])
    t2 = emzed.utils.stackTables([t1, t1])
    assert len(t2) == 2*len(t1)
