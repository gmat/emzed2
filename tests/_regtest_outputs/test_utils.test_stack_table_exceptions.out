detected incompatibilities:
       0/2 col(number=0, name=a): <type 'int'> vs <type 'str'>
       0/2 col(number=0, name=a): %d vs %s
tables 0 and 1 have different number columns (2 and 1)
column names do not fit:
  column  0: a vs c
detected incompatibilities:
       0/2 col(number=0, name=a): %4d vs %d
